### PR TITLE
Support for signed_commits

### PR DESCRIPTION
**Pull request by dennisbrouwer91**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -121,6 +121,12 @@ variable "readers" {
   description = "A list of GitHub teams that should have read access"
 }
 
+variable "require_signed_commits" {
+  type        = bool
+  default     = false
+  description = "Boolean to enable or disable signed commits requirement"
```
Comment (shoekstra) [inline variables.tf:127]:
```suggestion
  description = "Boolean to require signed commits"
```

**Review by shoekstra — CHANGES_REQUESTED** [review]

File: variables.tf [hunk 2]
```diff
@@ -121,6 +121,12 @@ variable "readers" {
   description = "A list of GitHub teams that should have read access"
 }
 
+variable "require_signed_commits" {
```
Comment (marwinbaumannsbp) [inline variables.tf:124]: Should this not be part of the branch_protection variable?

Comment (shoekstra) [reply variables.tf:124]: Good catch 👍  I believe it should as it's used in the `github_branch_protection` resource.

File: main.tf [hunk 3]
```diff
@@ -79,11 +79,12 @@ resource "github_team_repository" "readers" {
...
-  enforce_admins    = local.protection[count.index].enforce_admins
-  pattern           = local.protection[count.index].branch
-  push_restrictions = local.protection[count.index].push_restrictions
-  repository_id     = github_repository.default.name
+  count                  = length(local.protection)
+  enforce_admins         = local.protection[count.index].enforce_admins
+  pattern                = local.protection[count.index].branch
+  push_restrictions      = local.protection[count.index].push_restrictions
+  repository_id          = github_repository.default.name
+  require_signed_commits = local.protection[count.index].require_signed_commits
```
Comment (marwinbaumannsbp) [inline main.tf:87]: You have to change the local as well :-)

**Review by marwinbaumannsbp — DISMISSED** [review]
Make sure to create a new tag according to semantic versioning and make sure its tagged as a breaking change version

**Review by shoekstra — DISMISSED** [review]

**Comment by shoekstra** [comment]
> Make sure to create a new tag according to semantic versioning and make sure its tagged as a breaking change version

And by this, `0.4.0` 😉

File: main.tf [hunk 4]
```diff
@@ -8,11 +8,12 @@ locals {
...
-        enforce_admins    = config.enforce_admins
-        push_restrictions = config.push_restrictions
-        required_reviews  = config.required_reviews
-        required_checks   = config.required_checks
+        branch                 = branch
+        enforce_admins         = config.enforce_admins
+        push_restrictions      = config.push_restrictions
+        required_reviews       = config.required_reviews
+        required_checks        = config.required_checks
+        require_signed_commits = config.require_signed_commits
```
Comment (svanharmelen) [inline main.tf:16]: Please use correct alphabetical ordering (goes for all locations of course):
```suggestion
        branch                 = branch
        enforce_admins         = config.enforce_admins
        push_restrictions      = config.push_restrictions
        require_signed_commits = config.require_signed_commits
        required_reviews       = config.required_reviews
        required_checks        = config.required_checks
```

**Review by svanharmelen — CHANGES_REQUESTED** [review]

**Review by svanharmelen — DISMISSED** [review]

**Review by svanharmelen — APPROVED** [review]

**Pull request merged**